src/ADSB/ADSBTCPLink.cc

```diff
@@ -8,165 +8,217 @@
+        return;
+    }
 
-        ADSBVehicle::ADSBVehicleInfo_t adsbInfo;
-        adsbInfo.icaoAddress = icaoAddress;
+    const int msgType = line.at(4).digitValue();
+    if (msgType == ADSB::ADSBMessageType::Unsupported) {
+        qCDebug(ADSBTCPLinkLog) << "ADSB Invalid message type " << msgType;
+        return;
+    }
+
+    // Skip unsupported mesg types to avoid parsing
+    if ((msgType == ADSB::ADSBMessageType::SurfacePosition) || (msgType > ADSB::ADSBMessageType::SurveillanceId)) {
+        return;
+    }
 
-        switch (msgType) {
-        case 1:
-        case 5:
-        case 6:
+    qCDebug(ADSBTCPLinkLog) << "ADSB SBS-1" << line;
+
+    const QStringList values = line.split(QChar(','));
+    if (values.size() < 4) {
+        return;
+    }
+
+    bool icaoOk;
+    const uint32_t icaoAddress = values.at(4).toUInt(&icaoOk, 16);
+    if (!icaoOk) {
+        return;
+    }
+
+    ADSBVehicle::ADSBVehicleInfo_t adsbInfo;
+    adsbInfo.icaoAddress = icaoAddress;
+
+    switch (msgType) {
+        case ADSB::ADSBMessageType::IdentificationAndCategory:
+        case ADSB::ADSBMessageType::SurveillanceAltitude:
+        case ADSB::ADSBMessageType::SurveillanceId:
             _parseAndEmitCallsign(adsbInfo, values);
             break;
-        case 3:
+
+        case ADSB::ADSBMessageType::AirbornePosition:
             _parseAndEmitLocation(adsbInfo, values);
             break;
-        case 4:
+
+        case ADSB::ADSBMessageType::AirborneVelocity:
             _parseAndEmitHeading(adsbInfo, values);
             break;
-        }
+
+        default:
+            break;
     }
 }
 
-void ADSBTCPLink::_parseAndEmitCallsign(ADSBVehicle::ADSBVehicleInfo_t &adsbInfo, QStringList values)
+void ADSBTCPLink::_parseAndEmitCallsign(ADSBVehicle::ADSBVehicleInfo_t &adsbInfo, const QStringList &values)
 {
-    QString callsign = values[10].trimmed();
+    if (values.size() < 10) {
+        return;
+    }
+
+    const QString callsign = values.at(10).trimmed();
     if (callsign.isEmpty()) {
         return;
     }
 
     adsbInfo.callsign = callsign;
     adsbInfo.availableFlags = ADSBVehicle::CallsignAvailable;
+
     emit adsbVehicleUpdate(adsbInfo);
 }
 
-void ADSBTCPLink::_parseAndEmitLocation(ADSBVehicle::ADSBVehicleInfo_t &adsbInfo, QStringList values)
+void ADSBTCPLink::_parseAndEmitLocation(ADSBVehicle::ADSBVehicleInfo_t &adsbInfo, const QStringList &values)
 {
-    bool altOk, latOk, lonOk;
-    int modeCAltitude;
+    if (values.size() < 19) {
+        return;
+    }
 
-    QString altitudeStr = values[11];
     // Altitude is either Barometric - based on pressure, in ft
     // or HAE - as reported by GPS - based on WGS84 Ellipsoid, in ft
     // If altitude ends with H, we have HAE
     // There's a slight difference between Barometric alt and HAE, but it would require
     // knowledge about Geoid shape in particular Lat, Lon. It's not worth complicating the code
+    QString altitudeStr = values.at(11);
```

Comment:
Curious as to what the difference is here?

Reply:
When debugging it uses a Q_ASSERT to verify a valid index, otherwise it is exactly the same. Just an easy way to catch errors

Reply:
So does `[]` operator.